### DISCUSSION TITLE
Using Collins Dictionary

### DISCUSSION THREAD

**Discussion by soubhik-c** [soubhik-c/Buygame#2] [Announcements]
I gave the dictionary issue another thought. I think it might be best to use Scrabble dictionary as it’s well-established and already being used for tournaments.

It turns out there are two main versions (for English speaking world at least).

NASPA is for North America. Referred to as NWL2020 (most recent version). See here: [https://en.wikipedia.org/wiki/NASPA_Word_List](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FNASPA_Word_List&data=05%7C01%7Cschakraborty%40falcon.bentley.edu%7Cd7cbf5fc3a084547386f08db344f4242%7C9030beae3cfc4788a9e2130204ff1f10%7C0%7C0%7C638161287397213964%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=im9%2Fb7I1GIp6kbFFAeNSoFJTYAoyuXYtdKjEUI9t6iE%3D&reserved=0).

Collins is used in most English-speaking countries except the US and Canada. Referred to as SOWPODS. See here: [https://en.wikipedia.org/wiki/NASPA_Word_List](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FNASPA_Word_List&data=05%7C01%7Cschakraborty%40falcon.bentley.edu%7Cd7cbf5fc3a084547386f08db344f4242%7C9030beae3cfc4788a9e2130204ff1f10%7C0%7C0%7C638161287397213964%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=im9%2Fb7I1GIp6kbFFAeNSoFJTYAoyuXYtdKjEUI9t6iE%3D&reserved=0).

There are also some minor versions, like NSWL etc. for school Scrabble competitions etc. You can read more in the official [NASPA website](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fscrabbleplayers.org%2Fw%2FNSWL2020&data=05%7C01%7Cschakraborty%40falcon.bentley.edu%7Cd7cbf5fc3a084547386f08db344f4242%7C9030beae3cfc4788a9e2130204ff1f10%7C0%7C0%7C638161287397213964%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=kMhaNEPydgQ1tRmYqzhk4xPHS59OwtKzjn8cAZJoMGA%3D&reserved=0) if you are interested. But ultimately, as we will be running the game as an experiment in US, we would be better off using the official NWL2020 ([https://scrabbleplayers.org/w/NWL2020](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fscrabbleplayers.org%2Fw%2FNWL2020&data=05%7C01%7Cschakraborty%40falcon.bentley.edu%7Cd7cbf5fc3a084547386f08db344f4242%7C9030beae3cfc4788a9e2130204ff1f10%7C0%7C0%7C638161287397213964%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=cx6XgfZkCwpbJfaADgOumn0%2BC9hMFqRlWDailcj6dKg%3D&reserved=0)).

Normally, one must be a member to download it from the official website. But I found a GitHub page that collects Scrabble word lists: [https://github.com/scrabblewords/scrabblewords](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fscrabblewords%2Fscrabblewords&data=05%7C01%7Cschakraborty%40falcon.bentley.edu%7Cd7cbf5fc3a084547386f08db344f4242%7C9030beae3cfc4788a9e2130204ff1f10%7C0%7C0%7C638161287397213964%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Gyoyz1fxgfRw10jN2Wmns0L7%2FKMlScfZEwWlxUJxYr0%3D&reserved=0). For example, under words/North-American, there is NWL2020.txt, which is what I wanted (I’m attaching this file here). It has the words as well as short definitions, which is nice. It has 191,852 lines, which is exactly the number shown in NASPA wikipedia page ([https://en.wikipedia.org/wiki/NASPA_Word_List#Word_count](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FNASPA_Word_List%23Word_count&data=05%7C01%7Cschakraborty%40falcon.bentley.edu%7Cd7cbf5fc3a084547386f08db344f4242%7C9030beae3cfc4788a9e2130204ff1f10%7C0%7C0%7C638161287397213964%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=A7gy%2BjUy8Lb6CYHIqGeFFz69Q%2F8X5VPBreIYl5xDZcY%3D&reserved=0)). This tells me it’s accurate.

Let me know what you think.

Best,

Dr Gorkem Aksaray
Assistant Professor of Entrepreneurship
Trinity Business School, Trinity College Dublin

**Comment by soubhik-c** [comment]
[NWL2020.txt](https://github.com/soubhik-c/Buygame/files/11153860/NWL2020.txt)

Okay, let me work on incorporating this snapshot of the dictionary. We need to strip down the meaning portion and have only the word per line for it to work.

**Comment by soubhik-c** [comment]
<img width="1270" alt="Screenshot 2023-04-04 at 7 20 33 PM" src="https://user-images.githubusercontent.com/11516481/229944025-275c63dc-9f5e-446a-a53b-c1b5a48a1cf6.png">
@gaksaray with the new dictionary, player can get stuck with invalid words like attached screenshot. Imagine he/she "must" sell letters (no more than 8 can be held in the rack) and there is no possibility of "single" letter word(s) that can be sold... in this situation the player can NOT proceed at all.

**Reply by gaksaray** [reply]
I'm not very clear about the problem. Say I have 10 letters, and I *must* sell 2. Why can't I form a two-letter word to proceed?

**Reply by soubhik-c** [reply]
If in your rack lets say letters are mostly non-vowel letters and there is no wild card left, it becomes very hard to form a word. In few instances during my testing I had to sell one letter word twice for 2 letters to be sold out of 10. Probability wise it couldn't hit that situation very commonly but if it does there is "no way out" for the player.

**Reply by gaksaray** [reply]
I see. They would be selling these extra letters individually, which means they are penalized (as there is no compounding pips as you are selling a word) but a way out nevertheless. Am I understanding this correctly?

**Reply by soubhik-c** [reply]
That is correct, they will be able to proceed and finish the game.

**Reply by gaksaray** [reply]
Sounds good then

**Comment by gaksaray** [comment]
[nwl2020_wordlist.csv](https://github.com/soubhik-c/Buygame/files/11156907/nwl2020_wordlist.csv)

Here is the word list. I imported the original file to Stata using space delimiter and kept the first column.
